.ci/cico_rhche_compatibility_test.sh

```diff
@@ -5,129 +5,181 @@
-#Get last commit short hash from upstream che
-longHashUpstream=$(curl -s https://api.github.com/repos/eclipse/che/commits/master | jq .sha)
-shortHashUpstream=${longHashUpstream:1:7}
-
-#Get last commit short hash from rh-che branch 
-longHashDownstream=$(git log | grep -m 1 commit | head -1 | cut -d" " -f 2)
-shortHashDownstream=${longHashDownstream:0:7}
-
-#DOCKER_IMAGE_TAG is used for running tests
-#DOCKER_IMAGE_TAG_WITH_SHORTHASHES is used for tracking changes
-export DOCKER_IMAGE_TAG="upstream-check-latest"	
-export DOCKER_IMAGE_TAG_WITH_SHORTHASHES="upstream-check-$shortHashUpstream-$shortHashDownstream"
-export PROJECT_NAMESPACE=compatibility-check
-
-#set values needed for checking/creating PR
-RELATED_PR_TITLE="Update to $(echo $CHE_VERSION | cut -d'-' -f 1)"
-PR_BODY="Tracking changes for fixing compatibility with upstream $CHE_VERSION. This PR was created automatically by Jenkins from job $JOB_NAME"
-PR_HEAD="$BRANCH"
-PR_BASE="master"
-
-PULL_REQUESTS=$(curl -s https://api.github.com/repos/redhat-developer/rh-che/pulls?state=open | jq '.[].title')
-
-echo "Checking if PR exists."
-PR_EXISTS=1
-while read -r pr_title
-do
-  if [[ "$pr_title" == "\"$RELATED_PR_TITLE\"" ]]; then
-    echo "Pull request for tracking changes of version $CHE_VERSION has been already created."
-    PR_EXISTS=0
-    break
+set -e -m +x
+
+function checkPullRequest() {
+  PULL_REQUESTS=$(curl -s https://api.github.com/repos/redhat-developer/rh-che/pulls?state=open | jq '.[].title')
+  PR_EXISTS="false"
+  while read -r pr_title
+  do
+    if [[ "$pr_title" == "\"$RELATED_PR_TITLE\"" ]]; then
+      PR_EXISTS="true"
+      break
+    fi
+  done <<< "$PULL_REQUESTS"
+
+  if [[ "$PR_EXISTS" == "false" ]]; then
+    echo "false"
+  else
+    echo "true"
   fi
-done <<< "$PULL_REQUESTS"
+}
 
-#if PR does not exist, create it
-if [[ $PR_EXISTS -eq 1 ]]; then
+function createPullRequest() {
+  PR_BODY="Tracking changes for fixing compatibility with upstream $CHE_VERSION. This PR was created automatically by Jenkins from job $JOB_NAME"
+  PR_BASE="master"
   echo "Pull request for tracking changes of version $CHE_VERSION was not found - creating new one."
-	
-  #add changes (if there are some) and push branch
+  git push origin "$BRANCH" -f
+  curl -X POST -s -L -H "Content-Type: application/json" -u "${GITHUB_AUTH_STRING}" \
+       --data "{\"title\":\"$RELATED_PR_TITLE\", \"head\":\"$BRANCH\", \"base\":\"$PR_BASE\", \"body\":\"$PR_BODY\"}" \
+       https://api.github.com/repos/redhat-developer/rh-che/pulls
+  return_code=$?
+  if [ ! $return_code -eq 0 ]; then
+    echo $return_code > compatibility_test
+    exit $return_code
+  fi
+}
+
+function runCompatibilityTest() {
+  export USE_CHE_LATEST_SNAPSHOT="true"
+  export BASEDIR=$(pwd)
+  export DEV_CLUSTER_URL=https://devtools-dev.ext.devshift.net:8443/
+  export CHE_VERSION=$(curl -s https://raw.githubusercontent.com/eclipse/che/master/pom.xml | grep "^    <version>.*</version>$" | awk -F'[><]' '{print $3}')
```

Comment:
Another option here, if `yq` is installed in the CI system, is to use `xq`, which I believe comes with `yq`:

```
curl -s [url] | xq -r '.project.version'
```

Reply:
This is also news to me :D but thank you for the suggestion, will try it out